File: Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        vector<int> ret;
        int len = numbers.size(); 
        for( int i = 0; i <= len- 1; i++ ){
            for( int j = i+1; j <= len-1; j++ ){
                if( numbers[i] + numbers[j] == target ){
                    ret.push_back( i+1 );
                    ret.push_back( j+1 );
                }
            }
        }
        return ret; 
    }
};

File: LeetCode-Java/Two Sum.java
import java.util.Hashtable; 

public class Solution { 
	public int[] twoSum(int[] numbers, int target){ 
		int [] res = new int [2];
		int len = numbers.length; 
		Hashtable<Integer, Integer> ht = new Hashtable<Integer, Integer>();
		for( int i = 0; i < len; i++ ){
			Integer nn = ht.get(numbers[i]);
			if( nn == null ){
				Integer remainingIndex = ht.get( target - numbers[i] );
				if( remainingIndex != null ){
					if( i < remainingIndex ){
						res[0] = i+1;
						res[1] = remainingIndex+1;
					}else{
						res[0] = remainingIndex+1;
						res[1] = i+1;
					}
					return res;
				}
				ht.put(numbers[i], i);
			}else{
				if( nn < i && numbers[i] * 2 == target ){
					res[0] = nn+1;
					res[1] = i+1;
					return res;
				}
			}
		}

		return res;
	}
}


